From: Carlos Garnacho <carlosg@gnome.org>
Date: Sat, 1 Jul 2023 18:57:13 +0200
Subject: compositor: Do not repick after effects finish

This was a temporary fix until there was a better crossing event
delivery mechanism that accounted for actor changes beneath the pointer.
We nowadays have that, and don't seem to need this extra kick to get
crossing events triggered (and cursor changes, etc) when windows appear
or disappear under the pointer.

This commit is effectively a revert of commit
a64dba4d7aadadbab6abe1f09aac4611848842d4.

(cherry picked from commit 281d896f6ad30b114f17b0ac6eb1eaf6f394e956)

Origin: upstream, 44.6, commit:a74f79f20099403cb81991c3df659bcec35bc2bc
---
 src/compositor/meta-window-actor.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src/compositor/meta-window-actor.c b/src/compositor/meta-window-actor.c
index 48d4d02..f8c4604 100644
--- a/src/compositor/meta-window-actor.c
+++ b/src/compositor/meta-window-actor.c
@@ -686,11 +686,6 @@ meta_window_actor_after_effects (MetaWindowActor *self)
 {
   MetaWindowActorPrivate *priv =
     meta_window_actor_get_instance_private (self);
-  ClutterStage *stage;
-  ClutterSeat *seat;
-
-  stage = CLUTTER_STAGE (clutter_actor_get_stage (CLUTTER_ACTOR (self)));
-  seat = clutter_backend_get_default_seat (clutter_get_default_backend ());
 
   if (priv->needs_destroy)
     {
@@ -702,8 +697,6 @@ meta_window_actor_after_effects (MetaWindowActor *self)
       meta_window_actor_sync_visibility (self);
       meta_window_actor_sync_actor_geometry (self, FALSE);
     }
-
-  clutter_stage_repick_device (stage, clutter_seat_get_pointer (seat));
 }
 
 void
